core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java

```diff
@@ -33,40 +35,34 @@ public interface RdbEngineStrategy {
 
   String computeBooleanValue(boolean value);
 
-  void createNamespaceExecute(Connection connection, String fullNamespace) throws SQLException;
+  String[] createNamespaceExecuteSqls(String fullNamespace);
 
   String createTableInternalPrimaryKeyClause(
       boolean hasDescClusteringOrder, TableMetadata metadata);
 
-  void createTableInternalExecuteAfterCreateTable(
-      boolean hasDescClusteringOrder,
-      Connection connection,
-      String schema,
-      String table,
-      TableMetadata metadata)
-      throws SQLException;
+  String[] createTableInternalSqlsAfterCreateTable(
+      boolean hasDescClusteringOrder, String schema, String table, TableMetadata metadata);
 
-  void createMetadataTableIfNotExistsExecute(Connection connection, String createTableStatement)
-      throws SQLException;
+  String tryAddIfNotExistsToCreateTableSql(String createTableSql);
 
-  void createMetadataSchemaIfNotExists(Connection connection, String metadataSchema)
-      throws SQLException;
+  String[] createMetadataSchemaIfNotExistsSql(String metadataSchema);
 
-  void deleteMetadataSchema(Connection connection, String metadataSchema) throws SQLException;
+  boolean isCreateMetadataSchemaDuplicateSchemaError(SQLException e);
```

Comment:
It looks like we can unify this `isCreateMetadataSchemaDuplicateSchemaError()` and `isDuplicateUserError()` and `isDuplicateSchemaError()`?

Reply:
f36f6426